### PR TITLE
github actions for backend builds

### PR DESCRIPTION
**Pull request by nehaLohia27**
Signed-off-by: Neha Lohia <nehapithadiya444@gmail.com>

### CODE REVIEW

File: .github/workflows/backend-build.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      - github_actions # TODO delete this if ever merged to master/main
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: '1.17.0'
```
Comment (mcluseau) [inline .github/workflows/backend-build.yml:22]: we are at 1.17.3 now, if it's available here :)

Comment (jayunit100) [reply .github/workflows/backend-build.yml:22]: whoops just saw - ill make a pr to fix

Comment (jayunit100) [reply .github/workflows/backend-build.yml:22]: https://github.com/kubernetes-sigs/kpng/pull/98

**Comment by jayunit100** [comment]
big thanks to @FriedrichWilken and @nehaLohia27 for working together on this

**Pull request merged**